### PR TITLE
Enable autodetection params determine the pack position for different hands

### PR DESCRIPTION
**Pull request by beatrizleon**
## Proposed changes

Please explain the changes you made. Add pictures or videos to explain them better if appropriate.

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [x] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [x] Manually tested on hardware (if hardware specific or related).

## Documentation

- [x] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py [hunk 1]
```diff
@@ -96,12 +84,33 @@ def __init__(self, urdf_str=None, load=True, save=True):
                 thumb = True
             if is_lite and key.endswith("WRJ2"):
                 is_lite = False
+        hand_name = side + "_hand"
```
Comment (bdenoun) [inline sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]: Could we be using f-string instead?

Comment (bdenoun) [reply sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]:
```suggestion
        hand_name = f"{side}_hand"
```

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py [hunk 2]
```diff
@@ -96,12 +84,33 @@ def __init__(self, urdf_str=None, load=True, save=True):
...
+        
+        param = side+'_tip_sensors'
```
Comment (bdenoun) [inline sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]:
```suggestion
        param = f"{side}_tip_sensors"
```

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py [hunk 3]
```diff
@@ -96,12 +84,33 @@ def __init__(self, urdf_str=None, load=True, save=True):
...
+        while not rospy.has_param(param):
+            rospy.sleep(0.5)
+            rospy.loginfo(f"waiting for {str(param)}")
```
Comment (bdenoun) [inline sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]: I don't think you need the `str()` if you are using f-strings. But maybe there's a reason here?

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py [hunk 4]
```diff
@@ -96,12 +84,33 @@ def __init__(self, urdf_str=None, load=True, save=True):
...
+        # Check if hand has biotac 2p sensors
+        if tip_sensors_param.find('bt_2p') > -1:
+            tip_sensors = "bt_2p"
+        elif tip_sensors_param.find('bt_sp') > -1:
+            tip_sensors = "bt_sp"
+
+        param = side+'_hand_version'
+        while not rospy.has_param(param):
+            rospy.sleep(0.5)
+            rospy.loginfo(f"waiting for {str(param)}")
```
Comment (bdenoun) [inline sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]: Same here?

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py [hunk 5]
```diff
@@ -96,12 +84,33 @@ def __init__(self, urdf_str=None, load=True, save=True):
...
+        # load the tip_sensors from the parameter server after the hand has been auto-detected
+        tip_sensors_param = rospy.get_param(param)
+    
+        # Check if hand has biotac 2p sensors
+        if tip_sensors_param.find('bt_2p') > -1:
+            tip_sensors = "bt_2p"
+        elif tip_sensors_param.find('bt_sp') > -1:
+            tip_sensors = "bt_sp"
+
+        param = side+'_hand_version'
```
Comment (bdenoun) [inline sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]:
```suggestion
        param = f"{side}_hand_version"
```

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py [hunk 6]
```diff
@@ -96,12 +84,33 @@ def __init__(self, urdf_str=None, load=True, save=True):
...
+        # load the tip_sensors from the parameter server after the hand has been auto-detected
+        hand_version = rospy.get_param(param)
 
         rospy.logdebug(f"Found fingers (ff mf rf lf th) {str(first_finger)} {str(middle_finger)} " +
                        f"{str(ring_finger)} {str(little_finger)} {str(thumb)}")
         rospy.logdebug(f"is_lite: {str(is_lite)}")
-        rospy.logdebug(f"is_biotac: {str(is_biotac)}")
-        rospy.logdebug(f"Hand name: {str(hand_name)}")
+        rospy.logdebug(f"tip_sensors: {str(tip_sensors)}")
+        rospy.loginfo(f"hand_version: {str(hand_version)}")
```
Comment (bdenoun) [inline sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]: Not sure you need the str

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py [hunk 7]
```diff
@@ -111,7 +120,8 @@ def __init__(self, urdf_str=None, load=True, save=True):
                                   f'lf:={str(int(little_finger))}',
                                   f'th:={str(int(thumb))}',
                                   f'is_lite:={str(int(is_lite))}',
-                                  f'is_biotac:={str(int(is_biotac))}',
+                                  f'tip_sensors:={str(tip_sensors)}',
+                                  f'hand_version:={str(hand_version)}',
```
Comment (bdenoun) [inline sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]: Not sure you need the str

**Review by bdenoun — CHANGES_REQUESTED** [review]

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py (hunk 1 above)
Comment (beatrizleon) [reply sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]: Fixed

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py (hunk 2 above)
Comment (beatrizleon) [reply sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]: Fixed

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py (hunk 4 above)
Comment (beatrizleon) [reply sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]: Fixed

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py (hunk 5 above)
Comment (beatrizleon) [reply sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]: Fixed

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py (hunk 7 above)
Comment (beatrizleon) [reply sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]: Fixed

File: sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py (hunk 3 above)
Comment (beatrizleon) [reply sr_moveit_hand_config/scripts/sr_moveit_hand_config/generate_hand_srdf.py]: Fixed

**Review by rnzenha-s — APPROVED** [review]

**Comment by bdenoun** [comment]
Could we make a pass on the files to make sure we don't have `str` in f-strings? I still see some in some files

**Review by bdenoun — APPROVED** [review]

**Pull request merged**